cmd/pulumi-test-language/interface.go

```diff
@@ -568,88 +571,103 @@ func (eng *languageTestServer) RunLanguageTest(
 
 		sdkName := fmt.Sprintf("%s-%s", pkg, spec.Version)
 		sdkTempDir := filepath.Join(token.TemporaryDirectory, "sdks", sdkName)
-		_, err = os.Stat(sdkTempDir)
-		if err == nil {
-			// If the directory already exists then we don't need to regenerate the SDK
-			sdkArtifact, err := languageClient.Pack(sdkTempDir, artifactsDir)
-			if err != nil {
-				return nil, fmt.Errorf("sdk packing for %s: %w", pkg, err)
+		// Multiple tests might try to generate the same SDK at the same time so we need to be atomic here. There's two
+		// ways to do that. 1 is to generate to a temporary directory and then atomic rename it but Go say it doesn't
+		// support that, so option 2 we just lock around this section.
+		//
+		// TODO: This could probably be a per-sdk lock to be more fine grained and allow more parallelism.
```

Comment:
In general, any TODO we have in the code should have an associated tracking issue and linked to from the code.

Reply:
I'll raise one

Reply:
https://github.com/pulumi/pulumi/issues/16079